aicsimageio/writers/ome_zarr_writer.py

```diff
@@ -0,0 +1,406 @@
+                single_image_channel_colors_provided = False
+            else:
+                single_image_channel_colors_provided = True
+
+            if (
+                channel_colors[0] is not None
+                and isinstance(channel_colors[0], list)
+                and isinstance(channel_colors[0][0], int)
+            ):
+                single_image_channel_colors_provided = True
+
+        if channel_colors is None or single_image_channel_colors_provided:
+            channel_colors = [channel_colors] * num_images  # type: ignore
+
+        # Save image to zarr store!
+        mypath = pathlib.Path(uri)
+        # TODO handle overwrite scenario?
+        # shutil.rmtree(mypath)
+        # print(mypath)
+        store = parse_url(mypath, mode="w").store
+        # print(store)
+        root = zarr.group(store=store)
+        # print(root)
+        for scene_index in range(num_images):
+            image_data = data[scene_index]
+            pixelsize = physical_pixel_sizes[scene_index]
+            # TODO can this be deferred to inside of write_image?
+            # Assumption: if provided a dask array to save, it can fit into memory
+            if isinstance(image_data, da.core.Array):
+                image_data = data[scene_index].compute()  # type: ignore
+
+            # TODO image names must be unique!!!!!!
+            group = root.create_group(image_name[scene_index])
+            # TODO scaler might want to use different method for segmentations than raw
+            # TODO control how many levels of zarr are created
+            scaler = Scaler()
+            scaler.method = "nearest"
+            # TODO calculate these for desired downscaling factors
+            scaler.max_layer = 2
+            scaler.downscale = 2
+
+            # try to construct per-image metadata
+            ome_json = OmeZarrWriter.build_ome(
+                image_data.shape,
+                channel_names=channel_names[scene_index],  # type: ignore
+                channel_colors=channel_colors[scene_index],  # type: ignore
+                # this can be slow if going over all T values,
+                # might be better if user supplies the min/max?
+                channel_minmax=[
+                    (numpy.min(image_data[:, i, :]), numpy.max(image_data[:, i, :]))
+                    for i in range(image_data.shape[1])
+                ],
+            )
+
+            write_image(
+                image_data,
+                group,
+                chunks=(60, 256, 256),
+                scaler=scaler,
+                omero=ome_json,
+                axes=[
+                    {"name": "t", "type": "time", "unit": "millisecond"},
+                    {"name": "c", "type": "channel"},
+                    {"name": "z", "type": "space", "unit": "micrometer"},
+                    {"name": "y", "type": "space", "unit": "micrometer"},
+                    {"name": "x", "type": "space", "unit": "micrometer"},
+                ],
```

Comment:
this is "the big one" the chunks should be parameterizable / we should try to make nice assumptions for the user.

similar to the "chunk_dims" parameter on the readers maybe we have chunk dims parameter here? But I think people will also want to subdimension chunk so even more fine grain control

Reply:
user-defined output chunk sizes can be added later easily enough.  for now there is a hardcoded 16MB chunk size estimation in the code.